File: app/src/main/java/com/mirash/familiar/model/ICredentials.java
package com.mirash.familiar.model;

import com.mirash.familiar.tool.listener.Filterable;

/**
 * @author Mirash
 */
public interface ICredentials extends Filterable {
    String getTitle();

    String getLink();

    String getLogin();

    String getEmail();

    String getPhone();

    String getPassword();

    String getPin();

    String getDetails();

    int getId();

    int getPosition();
}

File: app/src/main/java/com/mirash/familiar/model/credentials/ICredentials.kt
package com.mirash.familiar.model.credentials

import com.mirash.familiar.tool.listener.Filterable
import kotlinx.serialization.Serializable

/**
 * @author Mirash
 */
@Serializable
sealed interface ICredentials : Filterable {
    val title: String?
    val link: String?
    val login: String?
    val email: String?
    val phone: String?
    val password: String?
    val pin: String?
    val details: String?
    val id: Long
    val position: Int
}
